tools/meteor-services/deploy.js

```diff
@@ -321,6 +323,206 @@ function canonicalizeSite(site) {
+    }).then(fiberHelpers.bindEnvironment(function() {
+      return pollForDeploy(pollingConfiguration, versionId, false);
+    })).await();
+  } else {
+    // Do a call to the version-status endpoint for the specified versionId
+    let versionStatusResult = deployRpc({
+      method: 'GET',
+      operation: 'version-status',
+      site: versionId,
+      expectPayload: [],
+      printDeployURL: false,
+    })
+    // We've done one call, so update the number of polling rounds
+    pollingRounds++;
+    let terminal = false;
+
+    //Check the details of the Version Status response and compare to last call
+    if(versionStatusResult &&
+      versionStatusResult.payload &&
+      versionStatusResult.payload.buildStatus) {
+      // Collect all of the statuses from the returned rpc call
+      const {buildStatus, deployStatus, isActive} = versionStatusResult.payload;
+
+      // If the build status has changed, we should report on that change to the user
+      // Allowable values for buildStatus are :
+      // ['missing', 'failed-temporarily', 'failed-permanently', 'success', 'building']
+      //    We will ignore the 'missing' buildStatus, and treat 'failed-temporarily'
+      //    the same as we do 'building', trusting the server to change the status.
+      if(currentStatus.buildStatus !== buildStatus) {
+        // Update our tracking variable for buildStatus
+        currentStatus.buildStatus = buildStatus;
+        // If we have reached a terminal buildStatus, handle it
+        if (terminalBuildStatuses.indexOf(buildStatus) >= 0) {
+          if(buildStatus == 'success') {
+            Console.info('Successfully built app image, initiating deploy');
+            // Reset the polling rounds to poll faster for a change in deployStatus
+            // Basically, we don't necessarily expect deploying to take a long time simply
+            // because building did.
+            pollingRounds = 1;
+          } else {
+            // Any other terminal status is failure, so report on it
+            Console.info(`Your app failed to build, with buildStatus: ${buildStatus}`);
+            // We also want to stop polling, since deploy cannot succeed with a failed build
+            terminal = true;
+          }
+        } else if (buildStatus !== 'building') {
+          // This will report if the buildStatus changes to a non-terminal, unexpected state
+          //    e.g. failed-temporarily
+          Console.info(`Build state changed to nonterminal state: ${buildStatus}`)
+        }
+      } else if(currentStatus.deployStatus !== deployStatus) {
+        // Deploy status is treated similarly to buildStatus above
+        currentStatus.deployStatus = deployStatus;
+        if (terminalDeployStatuses.indexOf(deployStatus) >= 0) {
+          if(deployStatus == 'success') {
+            Console.info('Your app has been deployed successfully.' +
+              ' Waiting for new version to be active...');
```

Comment:
So I think we should just make it so that when the app has deployed successfully, either it is active or it never will be.  I think that's more or less true now: the only way something will have deployed successfully without being active is if, before we poll, another version becomes active (a new deploy, or somebody clicks 'restore', or whatever).  So I think most of the time waiting for isActive will be trivial, and occasionally it will be infinite, and it will rarely (never?) be anything in between.

(You can see this in scheduler.go: we call PromoteProposedAppVersion immediately before we UpdateAppVersionDeployStatus to success.)

I think it might be interesting to note in the message if isActive is false (say something like "you've deployed successfully but it is no longer the active version", not sure how to phrase that) but blocking doesn't make sense to me, unless I'm missing something.